### PR TITLE
feat(Truncate): Add truncate wrapper

### PR DESCRIPTION
**Pull request by codecaaron**
## Truncate wrapper
`react-truncate` has been having some issues with getting the line width correct lately, and  has some limitations that make it unideal for general usage (no markup). 

This adds a wrapper for a replacement that has the same features for the most part: https://github.com/parsable/react-truncate-markup

Changes:
- Removes `react-truncate` in favor of `react-truncate-markup`.
- Adds a wrapper that simplifies some of the required syntax and room for simpler customization.

Note:
- Will add unit Tests if we're fine moving forward with this.  Foregoing before that.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.0a13a1.0<br/>@codecademy/gamut-templates@2.5.8-alpha.0a13a1.0<br/>@codecademy/gamut@2.15.1-alpha.0a13a1.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.3aeba5.0<br/>@codecademy/gamut-icons@2.2.1-alpha.3aeba5.0<br/>@codecademy/gamut-templates@2.5.8-alpha.3aeba5.0<br/>@codecademy/gamut@2.15.1-alpha.3aeba5.0<br/></pre>

File: packages/gamut/src/Truncate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}) => {
+  const content = <Element className={s.wrapper}>{children}</Element>;
+
+  if (!lines) {
+    return content;
+  }
+
+  return (
+    <TruncateMarkup
+      ellipsis={<span>...</span>}
```
Comment (codecaaron) [inline packages/gamut/src/Truncate/index.tsx:29]: Eventually the plan is to add some standardized variants here.  We use fade truncation a lot in the monolith, but that requires a level of color matching that seems out of scope for now.  Defaulting to standardized ellipsis eventually opening this up for customization.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b7b443489e46a4b6924f0--codecademy-styleguide.netlify.com>https://5e8b7b443489e46a4b6924f0--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b7b443489e46a4b6924f0>Deploy Logs</a></p>

File: packages/styleguide/__tests__/storybook-test.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import path from 'path';
 import initStoryshots, { renderOnly } from '@storybook/addon-storyshots';
 
+jest.mock('react-truncate-markup', () => (props: any) => props.children);
```
Comment (codecaaron) [inline packages/styleguide/__tests__/storybook-test.ts:4]: Unsure what to do here, the storyshot tests start to fail with ref usage, looking into a cleaner way to do this for now.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.ad9284.0<br/>@codecademy/gamut-templates@2.5.8-alpha.ad9284.0<br/>@codecademy/gamut@2.15.1-alpha.ad9284.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8b7e1bcc95e6330a962d4a--codecademy-styleguide.netlify.com>https://5e8b7e1bcc95e6330a962d4a--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8b7e1bcc95e6330a962d4a>Deploy Logs</a></p>

File: packages/styleguide/__tests__/storybook-test.ts (hunk 2 above)
Comment (jakemhiller) [reply packages/styleguide/__tests__/storybook-test.ts:4]: makes sense to me... doesn't seem like it would be compatible with testing like this

File: packages/gamut/src/Truncate/index.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  as: Element = 'span',
+  lines,
+  children,
+  onTruncate,
+}) => {
+  const content = <Element className={s.wrapper}>{children}</Element>;
+
+  if (!lines) {
+    return content;
+  }
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Truncate/index.tsx:25]: Is there a case where we wouldn't be passing `lines` in? Feels like that prop could be made required.

File: packages/gamut/src/Truncate/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.wrapper {
+  position: relative;
+  display: inline-block;
+  overflow: hidden;
+  visibility: visible;
+  background-color: inherit;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Truncate/styles.module.scss:6]: 🤔 what does this do?

File: packages/styleguide/__tests__/storybook-test.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/styleguide/__tests__/storybook-test.ts:4]: Bookmarking for later: https://github.com/parsable/react-truncate-markup/issues/28

File: packages/gamut/src/Truncate/index.tsx (hunk 3 above)
Comment (hmafzal) [reply packages/gamut/src/Truncate/index.tsx:25]: If it's truly not required, we should comment what the behavior is when it's not passed in the Props above

File: packages/gamut/src/Truncate/styles.module.scss (hunk 4 above)
Comment (codecaaron) [reply packages/gamut/src/Truncate/styles.module.scss:6]: Nice catch.  This is an artifact from my fade-ellipsis implementation, but have since removed. 

Removing for now.

File: packages/gamut/src/Truncate/index.tsx (hunk 3 above)
Comment (codecaaron) [reply packages/gamut/src/Truncate/index.tsx:25]: Yeah, I definitely like this less now after a night of sleep.  It was meant to be a cheap way to toggle it, without setting an arbitrary limit like 1000 lines.

What do y'all think about having a separate prop that does it? 
```tsx
<Truncate lines={2} shouldTruncate={!expanded} />
```

Or other option we can just have things be controlled in the parent component.

```tsx
const text = isExpanded ? whatever / <Truncate lines={2}>{whatever}</Truncate>
```

Fine with either.

Comment (hmafzal) [reply packages/gamut/src/Truncate/index.tsx:25]: Ahhh I see the use case. In that case, I think it's fine as is, and we can call it out in the comments (if not set, does not truncate).

FWIW react-truncate has the same api: https://www.npmjs.com/package/react-truncate

Comment (codecaaron) [reply packages/gamut/src/Truncate/index.tsx:25]: Heh I thats what incepted me 👀

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.10-alpha.cf3715.0<br/>@codecademy/gamut-templates@2.6.2-alpha.cf3715.0<br/>@codecademy/gamut@2.16.1-alpha.cf3715.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8c8ccaa1692690b38130ed--codecademy-styleguide.netlify.com>https://5e8c8ccaa1692690b38130ed--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8c8ccaa1692690b38130ed>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8c90a8cb0451025f32d7d3--codecademy-styleguide.netlify.com>https://5e8c90a8cb0451025f32d7d3--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8c90a8cb0451025f32d7d3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.10-alpha.75f602.0<br/>@codecademy/gamut-templates@2.6.2-alpha.75f602.0<br/>@codecademy/gamut@2.16.1-alpha.75f602.0<br/></pre>

File: packages/gamut/src/Truncate/index.tsx (hunk 3 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Truncate/index.tsx:27]: Oh, @codecaaron I wish I'd seen this before merging - they say `false` in their README.md but it looks like `undefined` will also do the trick. So this can become an optional prop as written in the story.
